File: recursion/egcd.py
def egcd(a, b):
	if a == 0:
		return (b, 0, 1)
	else:
		gcd, x, y = egcd(b % a, a)
		return(gcd, y - (b//a) * x, x)


gcd, a, b = egcd(588, 78)
print(gcd,a,b)

File: recursion/egcd.js
egcd = (a, b) => {
  if (b === 0) return { gcd: a, x: 1, y: 0 };

  const r = a % b;
  const tmp = egcd(b, r);

  return { gcd: tmp.gcd, x: tmp.y, y: tmp.x - Math.floor(a / b) * tmp.y };
};


console.log(egcd(588,78))